### ISSUE TITLE
General issues to be corrected

### ISSUE THREAD

**Issue by pedrohbraga** [QCBSRworkshops/workshop01#3]
General
- Generally, many slides of this workshop refer to features in RStudio and R, without pictures or snapshots of the window. This would require the instructor to constantly shift between the presentation and RStudio to demonstrate the topics being discussed - which can be confusing, especially in a remote setting. Special attention should be given to fix these issues, either by including more snapshots (static or even GIFs), by using a different set of tools, and/or reframing and updating the content of the slides.
- There are a few parts that lack explanation: e.g., the slide about error messages does not mention that error messages can be informative; the slide about objects tells us to avoid using `=` when assigning objects, but it does not say why;
- Examples should be more demonstrative: e.g. it is more helpful for participants to use code chunks and output to show what happens when you write objects that begin with letters or special characters or  what happens when you call objects without paying attention to cases, rather than giving them a list of rules (slide 32);

Specific
- [ ] **Important: scan and fix grammatical issues**;
- [ ] Remove abbreviations;
- [ ] Consistently use backticks to refer to R, packages, objects, functions and operators;
- [ ] Snapshots of RStudio and R should be updated to earlier versions, which include 
- [ ] Slides about arithmetic operations can all fit into one (incremental, if needed);
- [ ] The assignment operator in slide 31 one does not correspond to the one in R;
- [ ] Slides 38 and 39 (R Command Line Tips) should be reevaluated and blended;
- [ ] Slide 46 has a lot of text and would gain more if it had figures;
- [ ] Slide 47: standardize nomenclature with https://rc2e.com/datastructures

**Comment by beausoleilmo** [comment]
There is an item in the list that is not clear "Snapshots of RStudio and R should be updated to earlier versions, which include". Could you update the list to make it clearer?

**Comment by beausoleilmo** [comment]
## Objective of the workshop

As @pedrohbraga mentioned to me, here are the objectives of the workshop:  

Generally, this presentation would ideally have to be more aligned with the objectives of the workshop so participants can feel that they are confident with:
- [] (I) independently beginning to use R and RStudio; 
- [] (ii) identifying, inspecting, creating and being able to distinguish between vectors, matrices, data frames, arrays and lists; 
- [] (iii) performing basic data manipulation tasks (with square brackets, etc.) and data operations; 
- [] (iv) creating graphics; 
- [] (v) using the environment, the console and script windows; 
- [] (vi) reading and writing files to the disk (not necessarily in this order and not necessarily limited to this). As you can see, **the current format of the workshop needs to be worked to better fit these learning objectives**. 

Basically, what I would do is: prioritize 
- [] (i) spending a short amount of time planning on what each section of the workshop should be and what changes should be made, what should be completely reframed and what should be kept; 
- [] (ii) begin by adapting and fixing the major issues of this workshop, so it becomes presentable; and 
- [] (iii) only after this, add minor changes.

**Comment by beausoleilmo** [comment]
test

**Comment by beausoleilmo** [comment]
## Objectives of the workshop

As @pedrohbraga mentioned to me, here are the objectives of the workshop:

Generally, this presentation would ideally have to be more aligned with the objectives of the workshop so participants can feel that they are confident with:
- [ ] (I) independently beginning to use R and RStudio;
- [ ] (ii) identifying, inspecting, creating and being able to distinguish between vectors, matrices, data frames, arrays and lists;
- [ ] (iii) performing basic data manipulation tasks (with square brackets, etc.) and data operations;
- [ ] (iv) creating graphics;
- [ ] (v) using the environment, the console and script windows;
- [ ] (vi) reading and writing files to the disk (not necessarily in this order and not necessarily limited to this). As you can see, **the current format of the workshop needs to be worked to better fit these learning objectives** (as of 19 October 2020).

Basically, what I would do is: prioritize
- [ ] (i) spending a short amount of time planning on what each section of the workshop should be and what changes should be made, what should be completely reframed and what should be kept;
- [ ] (ii) begin by adapting and fixing the major issues of this workshop, so it becomes presentable; and
- [ ] (iii) only after this, add minor changes.

----------

## Pedagogical restucturing of the workshop:
**Overall presentation of the workshops (not limited to workshop 1)**
- [ ] **QCBS R workshop table of content**: Explain how the workshops (all 10 of them) were designed to be easy and pragmatic for non-programmers and was specifically designed for biologically oriented data analysis (but see **disclaimer**)
- [ ] Add a table of content of the workshops to make the workshops integrates with one another.
- [ ] **Learning objectives**: State the learning objectives at the beginning and at the end of the workshop (or what attendees of the workshop should get out at the end of the workshop).
- [ ] Add a slide with a series of questions to ask to questions to students:
- [ ] Questions could be about: 1. their name, 2. their university and their project (maybe supervisor), 3. what they are looking for to learn into the workshop series, and 4. encourage them in their project and try to find what, into the workshop series, fits their project. Maybe add a note into the presenters' note to say that the teacher should try to direct the relevant resources to the student (big maybe). Perhaps, encourage the teachers to look ahead and take a quick glance at all the other workshops to encourage the students to take the right workshop, tailored to what they are looking for.
- [ ] (Maybe) Add at the beginning (or at the end like it is at the moment) that there are many other resources if people are interested in to go further. Especially the [Intro to R](https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf) by W. N. Venables, D. M. Smith and the R Core Team is a deeper look into the **programming environment that is R**
- [ ] Design:
- [ ] please remove the image at the beginning of the workshop, (the image with the people and the swirly things, it refers to the *old* website design). The title slide cleaner and use only the QCBS logo with the title of the workshop.
- [ ] Clean the "About this workshop" slide. If you want to keep it, *only* put the "wiki", "Slides", "Slides" and "Script". Also, rename the "Slides", "Slides" badges so that they make more sense to someone less familiar with the icons.
- [ ] Add a survey at the end of the workshop so that students can input their ideas on how to improve the workshops. This was in the previous workshop and was helpful for future revamping sessions. Also, there might be a survey in the R coordinator drive that could be put there (https://drive.google.com/drive/folders/1Jt314Xb617Yw-Jkrrjlhl4MKhnKVVbeQ?usp=sharing).


**Statistics**
- [ ] **Global objectives of the workshops**: Question to @pedrohbraga: Are the workshops geared toward learning statistics or *only* learn R? What is the assumption or the prior requirements that students should have before coming to the QCBS R workshops?
- [ ] Add a "_Dislcaimer_" that would sound like this: The QCBS R workshops don't necessarily assume that you have a statistical background. But keep in mind that to make better decisions about the analyses you are planning to do, a background in statistics or experimental design is highly recommended. We do not pretend to solve all of your questions in how to analyse your particular dataset, but guide you through the process of analysis, interpretation and presentation of your data. We therefore assume that the data collection is going to be made or was made with a thoughtful process of data analysis, interpretation and presentation in mind. These workshops are designed to help you do the analysis, interpretation and presentation in R, but your experimental design is key to how the data will be analyzed, interpreted and presented. We highly recommend that you take a look at books about experimental design in you particular field of study and how the data collected is usually analyzed. (Maybe 1. Experimental Design and Data Analysis for Biologists, 2. An Introduction To Experimental Design And Statistics For Biology, or 3. Design and Analysis of Experiments with R or even take a look at this: https://stats.stackexchange.com/questions/1815/recommended-books-on-experiment-design)
- [ ] **Purpose of statistics**: State that statistics is roughly about 1. collecting (or simulating), 2. analyzing, 3. interpreting, and 4. effectively presenting you study and data. R is a tool to help you gain the insights about data you collect and test ideas with data that you can simulate! And it makes it available to you. Probability theory provides a crucial foundation for statistical thinking and reasoning (just look at \alpha, P-values, etc.).
- [ ] **What R changed and responsibilities**: Explain how before easy access to personal computers and free and accessible programs like R, scientists would need a trained statistical to test the assumptions of our models or know what are the preferred parameters values to draw certain types of graphs or explore the data to understand if they respect the model assumption. R makes it accessible to much more scientists and other institutions (governments, municipalities, NGOs and more) to do the basic tests needed to carry their research and allow them to do much more complex analysis that was out of reach before. This also comes with greater responsibilities in understanding, on your part, for example what statistical assumptions are underlying a t-test, a linear model, a PCA, a particular type of GLMM, etc.

**Why R**
- [ ] **Limits or R**: I don't agree that R is good for everything. Perhaps, after stating why R is too cool, state what are the limits to R (e.g., R can do spatial analysis, but if someone wants to do heavy work on map making and GIS analysis, QGIS is *much* better for that purpose [and there is a training on that at QCBS https://wiki.qcbs.ca/introgis]). **Data literacy** should be also encouraged as well as research integrity. R will not solve that in itself. Not sure if this is clear, but this could be added in the presenter's notes: "R is an open-source set of tools that has now been one of the world's leading statistical programming environment with a well developed community".

**Content**
- [ ] Update header or the R script (especially date of modification)
- [ ] Add sections using "`CMD+SHIFT+R`" (on Mac) and [I think...] `CTRL + SHIFT + R` (on Windows). This will allow students to see the outline of the R script with the "Show Document Outline" option in RStudio.
- [ ] Explain within the R script, how to add a comment (with `#`).
- [ ] Add Slides to make a break into the presentation. Add a note into the slide that the teacher *should* use the break time.
- [ ] **Inspiration**: take a look at the "[What is the most useful R trick?](https://stackoverflow.com/questions/1295955/what-is-the-most-useful-r-trick)" to inform the functions that should be learnt in the 1st workshop.
- [ ] **Built-in in R**: Add a note that R comes with **built in** constants (such as `pi` , `LETTERS`, `letters`, `month.abb`, `month.name`, see [this reference](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Constants.html))
- [ ] **Resources**:Encourage students to go to [Stack Overflow](https://stackoverflow.com). Especially, encourage them to ask questions with [reproducible examples](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example) and ask [better questions in general](https://stackoverflow.com/help/how-to-ask) on Stack Overflow. Perhaps not the best example out there, but [here is one question that I asked](https://stackoverflow.com/questions/45311490/is-it-possible-to-skip-na-values-in-operator).
- [ ] **Revision of the Challenges**:
- [ ] The pace of the challenges should be revised.
- [ ] It is not clear in the R script when the challenge ends. There could be a "end of challenge" tag or division to make this clear.
- [ ] Add a title to the challenges in the R script
- [ ] In general, challenges should include more `comments` about what is happening.
- [ ] In the script, add all the challenges (e.g., challenge 1 is not there, but state in the script that they were able to read the message in the challenge)
- [ ] Don't spend too much time on the "R as a calculator". This should take less than a minute. Perhaps, show to people how to make the appropriate symbols (there are always moments where people don't know how to make a `+`; `-`; `*`; `/`; `^`; `<`; `~`; `[]`; `()`, `{}`, `#`, etc.) Perhaps to fix this, tell them to try to make sure their keyboard is configured to be in english if possible. Otherwise, spend some time to show them how to find the characters on their keyboards. Also, explain what `1:10` means.
- [ ] Make sure that they know that there is a help to find keyboard shortcut in RStudio: (On Mac) "Tools" > "Keyboard Shortcuts Help" OR "`ALT + SHIFT + K`".
- [ ] Combine challenge 2 and 3
- [ ] Challenge 4: tell students that there exists preloaded values in R before the exercice.
- [ ] Challenge 5: use a more interesting number (in biology, nobody cares about Euler's number). Perhaps, just `result = 2+2`
- [ ] Challenge 6: Add a line stating that the number can be put elsewhere in the name (e.g. `result.8` or `result_8_times` or `result8`)
- [ ] Challenge 7: Why is there a `structure(c(1, 3, 5, 7)`) in challenge 7? Remove that.
- [ ] Challenge 8: Make sure to explain (probably in presenter's notes) that the type of subsetting (using `[]` or `[,]`) depends on the number of dimensions
- [ ] Challenge 9: It would be nice if the examples vectors (such as `char_vector`) are put back into the challenges. E.g., rewrite `char_vector <- c("blue", "red", "green")` into challenge 9 so that students who haven't run the line assigning the character values can still do the challenge. In my teacher's experience, it often happens that students were distracted or forgot to run the line of code 3 challenges before the one they are doing and now can't do the challenge because they have to find back the information needed to do the challenge.
- [ ] Challenge 10: Another (preferred) possibility is `my_df[, "num_sp"]`. The naming system for the object is pretty terrible. The workshop should contain a naming system that encourage students to continue doing. `my_df` should probably bee ore `fertilizer.data` (just thought about that, but perhaps explain that the `.data` is not as a `.pdf`, it is just part of the name of the object). Also, make `num_sp` more interesting by putting species names like `c("Statisticus revolusus", "Biodivertidus canadensis", "Statisticus revolusus", "Biodivertidus canadensis")` and naming the vector `species_names`. Why would one want to multiple "species number"? Perhaps make a new column called `abundance <- c(17, 23, 15, 7)` and make the `num_vector` called `plot_weights` or `sampling effort` or something to show that could be biologically relevant.
- [ ] Challenge 11: I don't see that the operator `:` was introduced. This should be added.
- [ ] Challenge 12: The title is not present into the script. Add that.
- [ ] Challenge 13: The challenge and the output don't match. Explain that the challenge is about creating a sequence (not with `:`, but with `seq()`, that increments `by 2`)
- [ ] Challenge 14: This challenge seems boring. Perhaps show them a function where there is a "default value is provided" that is not the one that we want as default. Therefore, students would have to dig the help page to find how to set the appropriate value. For example, `TukeyHSD()` has a default value for the confidence interval of 95% `conf.level = 0.95`. Let's say we want 97.5% or 99%. Something like that.

- [ ] **R History**: use the keyboard arrows to navigate the history of the commands that were done. If you use in RStudio on Mac the shortcut "`CMD + arrow-up`", you'll see a more of that history at a glance.

- [ ] Add a "show off how R is versatile" image:
![graph3](https://user-images.githubusercontent.com/15717151/96818158-e6c8de00-13ee-11eb-99e4-e02bf2025c52.png)


- [ ] **Simulating data**: Perhaps state that R is nice for simulating data and testing statistical ideas, or experimental designs, or play with data from the literature (that'd be nice also to add that into another future workshop or create one anew). Maybe show the t-distribution compared to the normal distribution.
Example (perhaps to complicated, but you get the idea):
```
# Sequence of number to draw the distributions
sequ = seq(-4,4, length.out = 100)

# Normal distribution
plot(sequ,y = dnorm(sequ), type = "l", ylim = c(0,1), main = "Normal and t-distributions", ylab = "Density")
# Area of shade (from https://www.r-bloggers.com/2012/06/shading-regions-of-the-normal-the-stanine-scale/)
alpha = 0.10
from.z.low <- -100
to.z.low <- qnorm(alpha/2)
from.z.up <- qnorm(1-(alpha/2))
to.z.up <- 100

s.x.low <- c(from.z.low, seq(from.z.low, to.z.low, 0.01), to.z.low)
s.y.low <- c(0, dnorm(seq(from.z.low, to.z.low, 0.01)), 0)
polygon(x = s.x.low, y = s.y.low, col="red")

s.x.up <- c(from.z.up, seq(from.z.up, to.z.up, 0.01), to.z.up)
s.y.up <- c(0, dnorm(seq(from.z.up, to.z.up, 0.01)), 0)
polygon(x = s.x.up, y = s.y.up, col="red")

# T-distribution
df = 1 # Setting the number of degrees of freedom for the t-distribution
lines(x = sequ,y = dt(sequ, df), type = "l", col = "red")

# Adding a legend to the plot
legend("topright",legend = c("Normal","t-distribution"),col = c(1,2), lty =1)
```

# Wiki
- [ ] In the wiki, it says "please contact the current series coordinators, listed on the main wiki page". It would be nice if there was a link to send an email or a link to state where to contact them!

### Final verification
- [ ] Use the R script the wiki and the presentation side-by-side to see what needs to be harmonized
- [ ] Make sure that the sections in the presentation, wiki and R script match (each header should be easy to find in each of the different tools)
